src/App.tsx

```diff
@@ -20,12 +37,15 @@ export const App: React.FC = () => {
                 id="search-query"
                 className="input"
                 placeholder="Type search word"
+                onChange={(event) => {
+                  setQuery(event.target.value);
+                }}
```

Comment:
Create a handler for this and trim a value inside it. Store only non-empty value.